rpcs3/rpcs3qt/main_window.cpp

```diff
@@ -766,15 +766,15 @@ void main_window::HandlePupInstallation(QString file_path)
 				if (dev_flash_tar_f.size() < 3)
 				{
 					gui_log.error("Error while installing firmware: PUP contents are invalid.");
-					QMessageBox::critical(this, tr("Failure!"), tr("Error while installing firmware: PUP contents are invalid."));
+					QMessageBox::critical(this, tr("Firmware Installation Failed"), tr("Firmware installation failed: Firmware could not be decompressed"));
```

Comment:
I think this message could be misleading, since the code hasn't attempted decompression yet.  If I saw that message, I'd suspect perhaps disk permission issues, or insufficient disk space/quota..

Perhaps something like 'Firmware installation failed: File hashes passed, however internal compressed image too small to decompress'

Reply:
MakeFile calls inflation routines in zlib, so it does decompress. It's called MakeFile but it actually compresses the data into a buffer and returns it wrapped as a fs::file by using container_stream.

Reply:
My bad.  Sorry, I should have looked at the whole file (and references).